spacerace-server/src/map/mod.rs

```diff
@@ -8,45 +11,75 @@ pub struct Map {
 }
 
 pub struct Obstacle {
-    pub position: Vec2,
-    pub size: Vec2,
+    pub polygon: Vec<Vec2>,
 }
 
 pub fn load_maps() -> HashMap<String, Map> {
     let mut maps = HashMap::new();
-    maps.insert(
-        "Gravity Map".to_string(),
-        Map {
-            name: "Gravity Map".to_string(),
-            gravity: 9.8,
-            obstacles: vec![
-                // Ground
-                Obstacle {
-                    position: Vec2::new(-180.0, -300.0),
-                    size: Vec2::new(1400.0, 10.0),
-                },
-                Obstacle {
-                    position: Vec2::new(-200.0, -200.0),
-                    size: Vec2::new(50.0, 50.0),
-                },
-                Obstacle {
-                    position: Vec2::new(0.0, 200.0),
-                    size: Vec2::new(50.0, 50.0),
-                },
-            ],
-        },
-    );
     maps.insert(
         "default_map".to_string(),
         Map {
             name: "default_map".to_string(),
             gravity: 0.0,
             obstacles: vec![Obstacle {
-                position: Vec2::new(-150.0, 100.0),
-                size: Vec2::new(75.0, 75.0),
+                polygon: vec![
+                    Vec2::new(-150.0, 100.0),
+                    Vec2::new(-75.0, 100.0),
+                    Vec2::new(-75.0, 175.0),
+                    Vec2::new(-150.0, 175.0),
+                    Vec2::new(-150.0, 100.0),
+                ],
             }],
         },
     );
 
+    let map = load_map_from_tiled("../test.json").unwrap();
+    maps.insert(map.name.clone(), map);
+
     maps
 }
+
+fn load_map_from_tiled(filename: &str) -> anyhow::Result<Map> {
+    let raw_map = tiled::Map::load_from_file(&PathBuf::from(filename))?;
+
+    // XXX load name from properties
+    // XXX load gravity from properties
+
+    let mut map = Map {
+        name: "tiled".to_string(),
+        gravity: 0.,
+        obstacles: vec![],
+    };
+
+    let objects =
+        find_first_object_layer(raw_map).ok_or(anyhow!("no object layer found in map"))?;
+    for object in objects {
+        tracing::warn!("found object {:?} {:?}", object.id, object.object_type);
+        if let tiled::ObjectType::Polygon(points) = &object.object_type {
+            tracing::warn!("object {:?} is polygon", object.id);
```

Comment:
Not at all important while developing locally but good habit to prefer structured logging over formatted strings e.g.

```rust
tracing::info!(game_id=?lobby[index].game_id, "Promoting game from lobby to active");
```

Reply:
Understood! These aren't staying around.